contracts/TransferLimitModule.sol

```diff
@@ -0,0 +1,445 @@
+        setManager();
+
+        require(domainSeparator == 0, "Domain Separator already set!");
+        // Greater than 1 hour and less than 1 year.
+        require(isValidTimePeriod(_timePeriod), "Invalid time period");
+        // In the range [1, safeThreshold - 1]
+        require(isValidThreshold(_threshold), "Invalid threshold");
+        require(_dutchxAddr != address(0), "Invalid dutchx address");
+
+        domainSeparator = keccak256(abi.encode(DOMAIN_SEPARATOR_TYPEHASH, this));
+
+        timePeriod = _timePeriod;
+        rolling = _rolling;
+        globalWeiCap = _globalWeiCap;
+        globalDaiCap = _globalDaiCap;
+        threshold = _threshold;
+        delegate = _delegate;
+        dutchx = DutchExchange(_dutchxAddr);
+
+        for (uint256 i = 0; i < tokens.length; i++) {
+            transferLimits[tokens[i]].limit = _transferLimits[i];
+        }
+    }
+
+    /// @dev Allows to update the transfer limit for a specified token. This can only be done via a Safe transaction.
+    /// @param token Token contract address.
+    /// @param transferLimit Transfer limit in smallest token unit.
+    function changeTransferLimit(address token, uint256 transferLimit)
+        public
+        authorized
+    {
+        transferLimits[token].limit = transferLimit;
+    }
+
+    /// @dev Updates the delegate. This can only be done via a Safe transaction.
+    /// @param _delegate New delegate, who is a non-owner account also allowed to make transfers.
+    function setDelegate(address _delegate)
+        public
+        authorized
+    {
+        delegate = _delegate;
+    }
+
+    /// @dev Returns if Safe transaction is a valid transfer limit transaction.
+    /// @param token Address of the token that should be transfered (0 for Ether)
+    /// @param to Address to which the tokens should be transfered
+    /// @param amount Amount of tokens (or Wei) that should be transfered
+    /// @param refund Gas to be refunded to relayer (gasPrice*gasLimit).
+    /// @param gasToken Token address (or 0 if ETH) that is used for the payment.
+    /// @param refundReceiver Address of receiver of gas payment (or 0 if tx.origin).
+    /// @param signatures Packed signature data ({bytes32 r}{bytes32 s}{uint8 v})
+    /// @return Returns if transaction can be executed.
+    function executeTransferLimit(
+        address token,
+        address to,
+        uint256 amount,
+        uint256 refund,
+        address gasToken,
+        address refundReceiver,
+        bytes memory signatures
+    )
+        public
+    {
+        require(to != address(0), "Invalid to address provided");
+        require(amount > 0, "Invalid amount provided");
+
+        bytes32 txHash = getTransactionHash(
+            token, to, amount,
+            refund, gasToken, refundReceiver,
+            nonce
+        );
+        require(checkSignatures(txHash, signatures), "Invalid signatures provided");
+        // Increase nonce and execute transaction.
+        nonce++;
```

Comment:
It would be more secure to increase the nonce before we call `checkSignatures`.
In this case it should not matter too much since right now this code doesn't support contract signatures, but as soon as this is the case this would be a security risk.